File: lib/types/metrics.ts
export interface KpiMetric {
  id: string;
  label: string;
  value: string;
  rawValue: number;
  delta: number;
  deltaUnit: string;
  direction: string;
  positive: boolean;
  series: { day: string; value: number }[];
  icon: string;
}

export interface PerformancePoint {
  day: string;
  runs: number;
  prevRuns: number;
}

export interface ViolationBreakdown {
  type: string;
  share: number;
  color: string;
}

export interface MetricsResponse {
  kpis: KpiMetric[];
  performanceSeries: PerformancePoint[];
  violationBreakdown: ViolationBreakdown[];
}

File: backend/src/audit_workbench/schemas/metrics.py
from pydantic import Field

from audit_workbench.schemas.common import CamelModel


class KpiSeriesPoint(CamelModel):
    day: str
    value: float


class KpiMetric(CamelModel):
    id: str
    label: str
    value: str
    raw_value: float
    delta: float
    delta_unit: str
    direction: str
    positive: bool
    series: list[KpiSeriesPoint]
    icon: str


class PerformancePoint(CamelModel):
    day: str
    runs: int
    prev_runs: int


class ViolationBreakdown(CamelModel):
    type: str
    share: float
    color: str


class HealthAlert(CamelModel):
    id: str
    severity: str
    title_key: str
    detail_key: str
    href: str | None = None


class MetricsResponse(CamelModel):
    kpis: list[KpiMetric]
    performance_series: list[PerformancePoint] = Field(serialization_alias="performanceSeries")
    violation_breakdown: list[ViolationBreakdown] = Field(serialization_alias="violationBreakdown")
    health_alerts: list[HealthAlert] = Field(
        default_factory=list, serialization_alias="healthAlerts"
    )
